### PR TITLE
Admin end points

### PR DESCRIPTION
**Pull request by AmateurBoy**
Need to reconcile the data in the Hall entity

### CODE REVIEW

File: apps/cinema-info-server/CinemaServer/Controllers/MainController.cs [hunk 1]
```diff
@@ -9,6 +8,7 @@ namespace CinemaServer.Controllers
 {
 
     [ApiController]
+
```
Comment (ImplementLife) [inline apps/cinema-info-server/CinemaServer/Controllers/MainController.cs]: ?

File: .gitignore [hunk 2]
```diff
@@ -1,3 +1,15 @@
 .idea
 .git
-.vs
\ No newline at end of file
+.vs
+/apps/cinema-info-server/CinemaServer/Migrations/20221013131507_Add_Hall_And_Session.cs
```
Comment (ImplementLife) [inline .gitignore:4]: You can ignore the folder

File: apps/cinema-info-server/CinemaServer/Controllers/AdminController.cs [hunk 3]
```diff
@@ -1,9 +1,10 @@
-﻿using CinemaServer.Entities;
+﻿using CinemaServer.Data.DTO;
+using CinemaServer.Entities;
 using CinemaServer.Services;
 using Microsoft.AspNetCore.Mvc;
 
 namespace CinemaServer.Controllers
-{
+{
```
Comment (ImplementLife) [inline apps/cinema-info-server/CinemaServer/Controllers/AdminController.cs]: ?

File: apps/cinema-info-server/CinemaServer/Data/Convertor/MovieConvertor.cs [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+using CinemaServer.Data.DTO.InterfaceDTO;
+using CinemaServer.Entities;
+
+namespace CinemaServer.Data.Convertor
+{
+    public class MovieConvertor : IBaseConvertor<Movie, DTOMainInfoMovie>
+    {
+        public Movie Convert(DTOMainInfoMovie dto)
+        {
+           Movie movie = new Movie();
```
Comment (ImplementLife) [inline apps/cinema-info-server/CinemaServer/Data/Convertor/MovieConvertor.cs]: Implementation?

File: apps/cinema-info-server/CinemaServer/Data/Entities/Session.cs [hunk 5]
```diff
@@ -0,0 +1,16 @@
+﻿using CinemaServer.Data.DTO.InterfaceDTO;
+using CinemaServer.Entities;
+
+namespace CinemaServer.Data.Entities
+{
+    public class Session : ISessionDTO
+    {
+        public int Id { get; set; }        
+        public DateTime ShowStartDate { get; set; }
```
Comment (ImplementLife) [inline apps/cinema-info-server/CinemaServer/Data/Entities/Session.cs:9]: Only time start

File: apps/cinema-info-server/CinemaServer/Services/CinemaService.cs [hunk 6]
```diff
@@ -1,52 +1,61 @@
...
+        public List<IMovieMainPageInfoDTO<ITagDTO>> MainCinema()
+        {   
+            MovieConvertor MC = new();            
+            var list = Context.Movies
+                .Include(x => x.Sessions)
+                .Include(x => x.Tags)
+                .Where(x => x.Sessions.Count > 0)
+                .Where(x => x.Sessions.Where(x=>x.ShowEndDate>DateTime.Now).Count()>0)
+                .ToList();
+            List<IMovieMainPageInfoDTO<ITagDTO>> ListDTO =new();
```
Comment (ImplementLife) [inline apps/cinema-info-server/CinemaServer/Services/CinemaService.cs]: add in converters the methods for converting lists

File: apps/cinema-info-server/CinemaServer/Data/DTO/InterfaceDTO/ISessionDTO.cs [hunk 7]
```diff
@@ -0,0 +1,9 @@
+﻿namespace CinemaServer.Data.DTO.InterfaceDTO
+{
+    public interface ISessionDTO
+    {
+        public int Id { get; set; }
+        public DateTime ShowStartDate { get; set; }
+        public DateTime ShowEndDate { get; set; }
```
Comment (ImplementLife) [inline apps/cinema-info-server/CinemaServer/Data/DTO/InterfaceDTO/ISessionDTO.cs:7]: Only Start date

File: apps/cinema-info-server/CinemaServer/Data/DTO/InterfaceDTO/IHallDTO.cs [hunk 8]
```diff
@@ -0,0 +1,14 @@
+﻿namespace CinemaServer.Data.DTO.InterfaceDTO
+{
+    public interface IHallDTO
+    {
+        public int Id { get; set; }
+        public string Name { get; set; }
+        public string Address { get; set; }
+        public int Rows { get; set; }
+        public int Columns { get; set; }
+        public int AllPlaces { get; set; }
```
Comment (ImplementLife) [inline apps/cinema-info-server/CinemaServer/Data/DTO/InterfaceDTO/IHallDTO.cs]: ??

**Review by ImplementLife — CHANGES_REQUESTED** [review]
Need update

File: .gitignore (hunk 2 above)
Comment (AmateurBoy) [reply .gitignore:4]: Completed

File: apps/cinema-info-server/CinemaServer/Controllers/MainController.cs (hunk 1 above)
Comment (AmateurBoy) [reply apps/cinema-info-server/CinemaServer/Controllers/MainController.cs]: Whitespace removal complete

File: apps/cinema-info-server/CinemaServer/Controllers/AdminController.cs (hunk 3 above)
Comment (AmateurBoy) [reply apps/cinema-info-server/CinemaServer/Controllers/AdminController.cs]: Whitespace removal complete

File: apps/cinema-info-server/CinemaServer/Data/Convertor/MovieConvertor.cs (hunk 4 above)
Comment (AmateurBoy) [reply apps/cinema-info-server/CinemaServer/Data/Convertor/MovieConvertor.cs]: Implementation

File: apps/cinema-info-server/CinemaServer/Services/CinemaService.cs (hunk 6 above)
Comment (AmateurBoy) [reply apps/cinema-info-server/CinemaServer/Services/CinemaService.cs]: Fulfilled

File: apps/cinema-info-server/CinemaServer/Data/DTO/InterfaceDTO/IHallDTO.cs (hunk 8 above)
Comment (AmateurBoy) [reply apps/cinema-info-server/CinemaServer/Data/DTO/InterfaceDTO/IHallDTO.cs]: image
Correct Table hall and interfaceDTO

File: apps/cinema-info-server/CinemaServer/Data/DTO/InterfaceDTO/IHallDTO.cs [hunk 9]
```diff
@@ -0,0 +1,13 @@
+﻿namespace CinemaServer.Data.DTO.InterfaceDTO
+{
+    public interface IHallDTO
+    {        public int Id { get; set; }
```
Comment (ImplementLife) [inline apps/cinema-info-server/CinemaServer/Data/DTO/InterfaceDTO/IHallDTO.cs]: ?

File: apps/cinema-info-server/CinemaServer/Services/InterfaceServices/IUpload.cs [hunk 10]
```diff
@@ -0,0 +1,7 @@
+﻿namespace CinemaServer.Services.Новая_папка
```
Comment (ImplementLife) [inline apps/cinema-info-server/CinemaServer/Services/InterfaceServices/IUpload.cs]: ?

File: apps/cinema-info-server/CinemaServer/Services/FileStorageService.cs [hunk 11]
```diff
@@ -0,0 +1,21 @@
...
+namespace CinemaServer.Services
+{
+    public class FileStorageService : IUpload
+    {
+        public string Upload(IFormFile File)
+        {   
+            DateTime dateTime = DateTime.Now;
+            string name = dateTime.Ticks.ToString();
+            name += File.FileName;
+            string filePut = @$"C:\Users\march\source\repos\Cinema\apps\cinema-info-server\CinemaServer\wwwroot\resources\images\{name}";
```
Comment (ImplementLife) [inline apps/cinema-info-server/CinemaServer/Services/FileStorageService.cs]: Move the path to the variable

File: apps/cinema-info-server/CinemaServer/Services/FileStorageService.cs [hunk 12]
```diff
@@ -0,0 +1,21 @@
+﻿using CinemaServer.Services.Новая_папка;
+
+namespace CinemaServer.Services
+{
+    public class FileStorageService : IUpload
+    {
+        public string Upload(IFormFile File)
+        {   
+            DateTime dateTime = DateTime.Now;
```
Comment (ImplementLife) [inline apps/cinema-info-server/CinemaServer/Services/FileStorageService.cs]: dateCreate will look best

File: apps/cinema-info-server/CinemaServer/Data/Entities/Hall.cs [hunk 13]
```diff
@@ -0,0 +1,20 @@
...
+namespace CinemaServer.Data.Entities
+{
+    public class Hall:IHallDTO
+    {
+        public int Id { get; set; }
+        public string Name { get; set; }
+        public string Address { get; set; } 
+        public int RowsCount { get; set; }
+        public int RowsPlaces { get; set; }
+        public int AllPlacesAreFree { get; set; }
```
Comment (ImplementLife) [inline apps/cinema-info-server/CinemaServer/Data/Entities/Hall.cs]: All places don't need it here..

File: apps/cinema-info-server/CinemaServer/Data/DTO/InterfaceDTO/IHallDTO.cs [hunk 14]
```diff
@@ -0,0 +1,13 @@
...
+        public string Name { get; set; }
+        public string Address { get; set; }
+        public int RowsCount { get; set; }
+        public int RowsPlaces { get; set; }
+        public int AllPlacesAreFree { get; set; }
```
Comment (ImplementLife) [inline apps/cinema-info-server/CinemaServer/Data/DTO/InterfaceDTO/IHallDTO.cs]: All places don't need it here..

File: apps/cinema-info-server/CinemaServer/Controllers/AdminController.cs (hunk 3 above)
Comment (ImplementLife) [reply apps/cinema-info-server/CinemaServer/Controllers/AdminController.cs]: hmmm< looks like as not

File: .gitignore [hunk 15]
```diff
@@ -1,3 +1,30 @@
...
+/apps/cinema-info-server/CinemaServer/Migrations/20221021153248_CorrectNaming.cs
+/apps/cinema-info-server/CinemaServer/Migrations/20221021153248_CorrectNaming.Designer.cs
+/apps/cinema-info-server/CinemaServer/Migrations/20221021180532_CorrectNaming2.Designer.cs
+/apps/cinema-info-server/CinemaServer/Migrations/20221021180532_CorrectNaming2.cs
+/apps/cinema-info-server/CinemaServer/Migrations/20221021180704_CorrectNaming3.cs
+/apps/cinema-info-server/CinemaServer/Migrations/20221021180704_CorrectNaming3.Designer.cs
+/apps/cinema-info-server/CinemaServer/wwwroot/resources/images/638019781421196925users.png
+/apps/cinema-info-server/CinemaServer/wwwroot/resources/images/638019828709493775avengers-infinity-war-phone-wallpaper.jpg
+/apps/cinema-info-server/CinemaServer/wwwroot/resources/images/638019837636038056blade-runner-2049-phone-wallpaper.jpg
+/apps/cinema-info-server/CinemaServer/wwwroot/resources/images/users.png
```
Comment (ImplementLife) [inline .gitignore:25]: remove from here all migrations and images

File: .gitignore [hunk 16]
```diff
@@ -1,3 +1,30 @@
...
+/apps/cinema-info-server/UnitTestServer/bin/Debug/net6.0/CinemaServer.deps.json
+/apps/cinema-info-server/UnitTestServer/bin/Debug/net6.0/appsettings.json
+/apps/cinema-info-server/UnitTestServer/bin/Debug/net6.0/appsettings.Development.json
+/apps/cinema-info-server/UnitTestServer
```
Comment (ImplementLife) [inline .gitignore:29]: UnitTestServer remove also

File: apps/cinema-info-server/CinemaServer/Services/CinemaService.cs [hunk 17]
```diff
@@ -1,52 +1,77 @@
...
+
     public class CinemaService:ICinemaService
     {
         private readonly AppDbContext Context;
+        public MovieConvertor movieConvertor = new();
+        public FileStorageService FileStorageService = new();
         public CinemaService (AppDbContext appDb)
         {
-            Context = appDb;
+            Context = appDb;
```
Comment (ImplementLife) [inline apps/cinema-info-server/CinemaServer/Services/CinemaService.cs]: ?

File: apps/cinema-info-server/CinemaServer/Data/Convertor/IBaseConvertor.cs [hunk 18]
```diff
@@ -0,0 +1,8 @@
+﻿namespace CinemaServer.Data.Convertor
+{
+    public interface IBaseConvertor<Base,DTO>
+    {
+        public Base Convert(DTO dto);
+        public DTO Convert(Base bases);
```
Comment (ImplementLife) [inline apps/cinema-info-server/CinemaServer/Data/Convertor/IBaseConvertor.cs:8]: Add here methods for converting Arrays of objects

File: apps/cinema-info-server/CinemaServer/Data/Convertor/MovieConvertor.cs [hunk 19]
```diff
@@ -0,0 +1,47 @@
...
+
+        public DTOMainInfoMovie Convert(Movie movie)
+        {            
+            DTOMainInfoMovie dTOMovie = new();
+            dTOMovie.Id = movie.Id;
+            dTOMovie.Name = movie.Name;
+            dTOMovie.NameImg = movie.NameImg;           
+            return dTOMovie;
+        }
+        public List<Movie> ConvertToList(List<DTOMainInfoMovie> dtolist)
```
Comment (ImplementLife) [inline apps/cinema-info-server/CinemaServer/Data/Convertor/MovieConvertor.cs]: Can be named just Convert

File: apps/cinema-info-server/CinemaServer/Data/Convertor/IBaseConvertor.cs (hunk 18 above)
Comment (ImplementLife) [reply apps/cinema-info-server/CinemaServer/Data/Convertor/IBaseConvertor.cs:8]: Can be named just Convert)

**Review by ImplementLife — CHANGES_REQUESTED** [review]
oooh< please update)

File: apps/cinema-info-server/CinemaServer/Data/DTO/InterfaceDTO/IHallDTO.cs (hunk 8 above)
Comment (ImplementLife) [reply apps/cinema-info-server/CinemaServer/Data/DTO/InterfaceDTO/IHallDTO.cs]: All places don't need here

**Review by ImplementLife — APPROVED** [review]
Ok

**Pull request merged**